src/com/vivian/java/leetcode/CCI4_7FirstCommAscentor.java

```diff
@@ -0,0 +1,53 @@
+package com.vivian.java.leetcode;
+/*Design an algrithm and write code to find the first common sncester
+ of two nodes in a Binary Tree. Aviod storing additional nodes in a data structure.
+ NOTE:This is not necessarily a BST.
+
+ Clarification: 1, return null if both nodes are not in the tree(OK).
+                2, return ascentor one,if one node is another's ascentor.
+
+ EdgeCase: 1,both nodes in the left side.
+           2,both nodes in the right side.
+           3,both nodes in different side.
+Approach: if both nodes on the left side Recursively left to find the common ascentor.
+          same way if both nodes on the right side.
+          if they're in diff side,Recursive the root,
+          until they are no longer on the same side,return the first node(ascentor)split them .
+
+complexity:o(n) since each node visited once
+
+*/
+public class CCI4_7FirstCommAscentor {
```

Comment:
Typo Comm -> Common, Ascentor -> Ancestor
Also, you can rename it to "CCI4_7LowestCommonAncestor"

Reply:
ok